File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/sensor/Sensor.java
package gov.doc.ntia.sigmf.ext.global.sensor;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import gov.doc.ntia.sigmf.ext.global.core.Antenna;
import gov.doc.ntia.sigmf.ext.global.core.HardwareSpec;
import gov.doc.ntia.sigmf.ext.global.environment.Environment;
import gov.doc.ntia.sigmf.ext.global.location.Location;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;


@JsonInclude(JsonInclude.Include.NON_NULL)
public class Sensor  implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value = "id", required = true)
    protected String id;

    @JsonProperty(value = "sensor_spec", required = false)
    protected HardwareSpec sensorSpec;

    @JsonProperty(value="ntia-core:antenna", required = false)
    protected Antenna antenna;

    @JsonProperty(value="preselector", required = false)
    protected Preselector preselector;

    @JsonProperty(value="signal_analyzer", required = false)
    protected SignalAnalyzer signalAnalyzer;

    @JsonProperty(value="computer_spec", required = false)
    protected HardwareSpec computerSpec;

    @JsonProperty(value="mobile", required=false)
    protected Boolean mobile;

    @JsonProperty(value="location", required = false)
    protected Location location;

    @JsonProperty(value="environment", required = false)
    protected Environment environment;

    public HardwareSpec getSensorSpec() {
        return sensorSpec;
    }

    public void setSensorSpec(HardwareSpec sensorSpec) {
        this.sensorSpec = sensorSpec;
    }

    public void setSignalAnalyzer(SignalAnalyzer signalAnalyzer) {
        this.signalAnalyzer = signalAnalyzer;
    }

    public HardwareSpec getComputerSpec() {
        return computerSpec;
    }

    public void setComputerSpec(HardwareSpec computerSpec) {
        this.computerSpec = computerSpec;
    }

    public Boolean getMobile() {
        return mobile;
    }

    public void setOtherFields(Map<String, Object> otherFields) {
        this.otherFields = otherFields;
    }

    public Boolean isMobile() {
        return mobile;
    }

    public void setMobile(Boolean mobile) {
        this.mobile = mobile;
    }

    protected Map<String, Object> otherFields = new HashMap<>();

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public Antenna getAntenna() {
        return antenna;
    }

    public void setAntenna(Antenna antenna) {
        this.antenna = antenna;
    }

    public Preselector getPreselector() {
        return preselector;
    }

    public void setPreselector(Preselector preselector) {
        this.preselector = preselector;
    }


    public SignalAnalyzer getSignalAnalyzer() {
        return signalAnalyzer;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public Environment getEnvironment() {
        return environment;
    }

    public void setEnvironment(Environment environment) {
        this.environment = environment;
    }

    @JsonAnyGetter
    public Map<String, Object> getOtherFields(){
        return otherFields;
    }

    @JsonAnySetter
    public void add(String key, Object value){
        otherFields.put(key, value);
    }

}

File: Csharp/sigmf/sigmf/ext/global/sensor/Sensor.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace sigmf
{
    [Serializable()]
    public class Sensor
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("id")]
        protected string id;

        [JsonProperty("sensor_spec")]
        protected HardwareSpec sensorSpec;

        [JsonProperty("antenna")]
        protected Antenna antenna;

        [JsonProperty("preselector")]
        protected Preselector preselector;

        [JsonProperty("signal_analyzer")]
        protected SignalAnalyzer signalAnalyzer;

        [JsonProperty("computer_spec")]
        protected HardwareSpec computerSpec;

        [JsonProperty("mobile")]
        protected bool mobile;

        protected Dictionary<string, Object> otherfields = new Dictionary<string, Object>();

        public string Id { get => id; set => id = value; }
        public HardwareSpec SensorSpec { get => sensorSpec; set => sensorSpec = value; }
        public Antenna Antenna { get => antenna; set => antenna = value; }
        public Preselector Preselector { get => preselector; set => preselector = value; }
        public SignalAnalyzer SignalAnalyzer { get => signalAnalyzer; set => signalAnalyzer = value; }
        public HardwareSpec ComputerSpec { get => computerSpec; set => computerSpec = value; }
        public bool Mobile { get => mobile; set => mobile = value; }

        public Dictionary<string, object> getOtherFields()
        {
            return otherfields;
        }
        public void add(string key, object value)
        {
            otherfields.Add(key, value);
        }
    }
}